autogpt_platform/backend/schema.prisma

```diff
@@ -272,11 +385,220 @@ model UserBlockCredit {
   blockId String?
   block   AgentBlock? @relation(fields: [blockId], references: [id])
 
+  // We need to be able to associate a credit transaction with an agent
+  executedAgentId      String?
+  executedAgentVersion Int?
+  executedAgent        Agent?  @relation(fields: [executedAgentId, executedAgentVersion], references: [id, version])
+
+  // We need to be able to associate a credit transaction with a store listing
+  storeListingId String?
+  storeListing   StoreListing? @relation(fields: [storeListingId], references: [id])
+
   amount Int
   type   UserBlockCreditType
 
-  isActive Boolean @default(true)
-  metadata Json?
+  isActive         Boolean         @default(true)
+  metadata         Json?
+  UserAccounting   UserAccounting? @relation(fields: [userAccountingId], references: [id])
+  userAccountingId String?
 
   @@id(name: "creditTransactionIdentifier", [transactionKey, userId])
 }
+
+// This is the top level accounting table for a user
+model UserAccounting {
```

Comment:
This is meant to be the table that is used to get all financial information about a user.